### PR TITLE
Add Suggestions Database Schema Version

### LINKED ISSUE #1590: When using Enso through the IDE, the file system is accessed very frequently
### General Summary
While the IDE is running, a Java process in the background (likely part of the engine) accesses the file system many times per second, even when there is no user interaction. This is not ideal for multiple reasons:
1. It slows down every other file system access. (especially on hard drives)
2. It increases the energy consumption. (especially on hard drives)
3. Hard drives can be damaged more easily while they are running. This is most relevant on laptops.

### Steps to Reproduce
Start the IDE and wait until startup is finished. A tool like `fatrace` will show that the file system is accessed many times per second.

### Expected Result
The disk should only be used infrequently and only for tasks where it is necessary, like backups.

### Actual Result
The disk is accessed many times per second. The following is a representative excerpt of the output produced by `fatrace`:

```
$ sudo fatrace
...
java(161558): W /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db-shm
java(161558): R /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db
java(161558): R /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db
java(161558): W /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db-wal
java(161558): O /home/s9ferech/enso/projects/Unnamed/.enso
java(161558): C /home/s9ferech/enso/projects/Unnamed/.enso
java(161558): W /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db-wal
java(161558): R /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db-wal
java(161558): W /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db
java(161558): R /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db-wal
java(161558): D /home/s9ferech/enso/projects/Unnamed/.enso
java(161558): CW (deleted)
java(161558): CW (deleted)
java(161558): D /home/s9ferech/enso/projects/Unnamed/.enso
java(161558): CW /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db
java(161558): O /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db
java(161558): R /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db
java(161558): R /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db
java(161558): + /home/s9ferech/enso/projects/Unnamed/.enso
java(161558): O /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db-wal
java(161558): + /home/s9ferech/enso/projects/Unnamed/.enso
java(161558): WO /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db-shm
java(161558): W /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db-shm
java(161558): W /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db-shm
java(161558): R /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db
java(161558): R /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db
java(161558): W /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db-wal
java(161558): CO /home/s9ferech/enso/projects/Unnamed/.enso
java(161558): W /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db-wal
java(161558): R /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db-wal
java(161558): W /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db
java(161558): R /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db-wal
java(161558): D /home/s9ferech/enso/projects/Unnamed/.enso
java(161558): CW (deleted)
java(161558): CW (deleted)
java(161558): D /home/s9ferech/enso/projects/Unnamed/.enso
java(161558): CW /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db
java(161558): O /home/s9ferech/enso/projects/Unnamed/.enso/suggestions.db
...
```

### Enso Version

```
$ ./dist/client/enso-linux-0.0.0.AppImage --version
Frontend:
    Version:  1.0.0
    Build:    ba0856e
    Electron: 11.1.1
    Chrome:   87.0.4280.88

Backend:
    Enso Project Manager
    Version:    0.2.6
    Built with: scala-2.13.5 for GraalVM 21.0.0.2
    Built from: enso-0.2.6 @ aec61292c467c3c52e2259ac9e8cd03670c77f15
    Built on:   Linux (amd64)
```

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

close #1590 

Changelog:
- feat: initialize suggestions database based on a schema version
- update: set SQLite journaling mode to memory

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/main/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/main/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guides.
- [x] All documentation and configuration conforms to the [markdown](https://github.com/enso-org/enso/blob/main/docs/style-guide/markdown.md) and [YAML](https://github.com/enso-org/enso/blob/main/docs/style-guide/yaml.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

File: engine/language-server/src/main/scala/org/enso/languageserver/boot/resource/RepoInitialization.scala [hunk 1]
```diff
@@ -26,31 +34,113 @@ class RepoInitialization(
...
+      _ <- suggestionsRepoInit
+      _ <- versionsRepoInit
     } yield InitializationComponent.Initialized
 
   private def suggestionsRepoInit: Future[Unit] = {
-    val initAction = suggestionsRepo.init
+    val initAction =
+      for {
+        _ <- Future {
+          log.info("Initializing suggestions repo.")
```
Comment (iamrecursion) [inline engine/language-server/src/main/scala/org/enso/languageserver/boot/resource/RepoInitialization.scala]: Where is it being initialized? You need to log actually useful information!

**Review by iamrecursion — CHANGES_REQUESTED** [review]
Actual fix looks good, but the log messages are lacking again.

File: engine/language-server/src/main/scala/org/enso/languageserver/boot/resource/RepoInitialization.scala (hunk 1 above)
Comment (4e6) [reply engine/language-server/src/main/scala/org/enso/languageserver/boot/resource/RepoInitialization.scala]: The source class is included in the log entry.
```
[info] [2021-04-28T07:41:24.949Z] [org.enso.languageserver.boot.resource.RepoInitialization] Initializing suggestions repo.
```

Comment (4e6) [reply engine/language-server/src/main/scala/org/enso/languageserver/boot/resource/RepoInitialization.scala]: Included the database file being initialized :+1:

**Review by iamrecursion — APPROVED** [review]

**Pull request merged**